### PR TITLE
Refactor add messages

### PR DESCRIPTION
**Pull request by typfel**
## What's new in this PR?

Method for appending messages and creating generic messages have been re-factored to be more composable.

### CODE REVIEW

File: Source/Model/Message/ZMClientMessage.m [hunk 1]
```diff
@@ -237,7 +237,7 @@ - (void)expire
 - (void)resend
 {
     if (self.genericMessage.hasEdited) {
-        NOT_USED([ZMMessage edit:self newText:self.textMessageData.messageText]);
+        NOT_USED([ZMMessage edit:self newText:self.textMessageData.messageText mentions:@[] fetchLinkPreview:YES]); // TODO jacob why can't we just resend the message?
```
Comment (mikeger) [inline Source/Model/Message/ZMClientMessage.m]: Seems like a todo, or not necessary any more?

File: Source/Utilis/Protos/ZMGenericMessage+Helper.swift [hunk 2]
```diff
@@ -416,6 +384,55 @@ public extension ZMOtrAssetMeta {
...
+    
+    public func setEphemeralContent(on builder: ZMEphemeralBuilder) {
+        builder.setAsset(self)
+    }
+    
+}
+
+extension ZMImageAsset: EphemeralMessageContentType {
+    
+//    public static func imageAsset(mediumProperties: ZMIImageProperties, processedProperties: ZMIImageProperties, encryptionKeys: ZMImageAssetEncryptionKeys, format: ZMImageFormat) -> ZMImageAsset {
```
Comment (mikeger) [inline Source/Utilis/Protos/ZMGenericMessage+Helper.swift]: Can be removed?

File: Tests/Source/Model/Conversation/ZMConversationTests+Messages.swift [hunk 3]
```diff
@@ -118,31 +118,13 @@ class ZMConversationMessagesTests: ZMConversationTestsBase {
...
-    {
-        // given
-        let conversation = ZMConversation.insertNewObject(in: self.uiMOC)
-        conversation.remoteIdentifier = UUID()
-        let start = self.uiMOC.insertedObjects
-    
-        // when
-        var message: Any? = nil
-        self.performIgnoringZMLogError {
-            message = conversation.appendMessage(withText: nil)
```
Comment (mikeger) [inline Tests/Source/Model/Conversation/ZMConversationTests+Messages.swift]: Good that we don't need this kind of test any more 👍

File: Source/Model/Message/ZMClientMessage.m (hunk 1 above)
Comment (typfel) [reply Source/Model/Message/ZMClientMessage.m]: Well I still don't why we re-generate the edited when re-sending it?

Comment (typfel) [reply Source/Model/Message/ZMClientMessage.m]: Let's try removing it and see it what happens :)

Comment (typfel) [reply Source/Model/Message/ZMClientMessage.m]: After talking to @daehn I'll remove this in a separate PR

**Pull request merged**